engines/sci/parser/vocabulary.cpp

```diff
@@ -288,12 +288,12 @@ bool Vocabulary::loadAltInputs() {
 		AltInput t;
 		t._input = data;
 
-		uint32 l = strlen(data);
+		uint32 l = Common::strnlen(data, data_end - data);
```

Comment:
the strlen()s inside checkAltInput() inside vocabulary.cpp aren't changed too?

As I said - I would prefer to error() out on corrupted resources. Because those would cause issues down the line anyway (unless there are games w/ actual corrupted resources that are still playable). That way the user would know about it immediately.